### PR TITLE
Add custom user

### PR DESCRIPTION
**Pull request by ubershloder**
Launching app with custom ruby user would be a better practice https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#user

### Motivation / Background

This Pull Request has been created because [REPLACE ME]

### Detail

Dockerfile.tt only

### Checklist

Before submitting the PR make sure the following are checked:

* [x] This Pull Request is related to one change. Changes that are unrelated should be opened in separate PRs.
* [x] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`
* [x] Tests are added or updated if you fix a bug or add a feature.
* [x] CHANGELOG files are updated for the changed libraries if there is a behavior change or additional feature. Minor bug fixes and documentation changes should not be included.

### CODE REVIEW

File: railties/lib/rails/generators/rails/app/templates/Dockerfile.tt [hunk 1]
```diff
@@ -70,9 +70,19 @@ RUN apt-get update -qq && \
 COPY --from=build /usr/local/bundle /usr/local/bundle
 COPY --from=build /rails /rails
 
+# add custom user
+RUN useradd ruby
+
+USER ruby:ruby
+
+COPY --from=builder --chown=ruby:ruby /usr/local/bundle /usr/local/bundle
```
Comment (byroot) [inline railties/lib/rails/generators/rails/app/templates/Dockerfile.tt:78]: It's totally wasteful to chown in another layer.

File: railties/lib/rails/generators/rails/app/templates/Dockerfile.tt [hunk 2]
```diff
@@ -70,9 +70,19 @@ RUN apt-get update -qq && \
 COPY --from=build /usr/local/bundle /usr/local/bundle
 COPY --from=build /rails /rails
 
+# add custom user
+RUN useradd ruby
```
Comment (byroot) [inline railties/lib/rails/generators/rails/app/templates/Dockerfile.tt:74]: Not sure I like the name `ruby`. Might be confusing. I'd rather use something like `appuser`

File: railties/lib/rails/generators/rails/app/templates/Dockerfile.tt (hunk 1 above)
Comment (ubershloder) [reply railties/lib/rails/generators/rails/app/templates/Dockerfile.tt:78]: you mean wasteful to have 2 chowns in 2 copy commands?

File: railties/lib/rails/generators/rails/app/templates/Dockerfile.tt (hunk 2 above)
Comment (ubershloder) [reply railties/lib/rails/generators/rails/app/templates/Dockerfile.tt:74]: agree here

File: railties/lib/rails/generators/rails/app/templates/Dockerfile.tt (hunk 1 above)
Comment (ubershloder) [reply railties/lib/rails/generators/rails/app/templates/Dockerfile.tt]: also updated, agree here, didn't notice it in the first place

**Comment by zzak** [comment]
Can we name the user `rails`? And would you please squash your commits? Thanks!

**Review by byroot — APPROVED** [review]
I just tested this locally, seem to work well.

I don't have a strong opinion on the name, I guess `rails` isn't bad.

And yes, please squash your commits.

**Comment by ubershloder** [comment]
Maybe `railsapp` or `railsuser`? Or i can keep the one we have now

**Comment by byroot** [comment]
Ok, I'll merge as is. If someone feel strongly about the user name, please open a PR and we can discuss it, but ultimately I think it's not that important.

**Comment by zzak** [comment]
Rename in #47385 :pray:

**Comment by dhh** [comment]
Seeing a permission error when using this together with a logger writing to a file. Probably need to chown the whole /rails directory as well, no?

**Comment by byroot** [comment]
@ddh what's the error exactly? Is the directory mounted or something?

And what's the directory permissions if you exec inside the container?

**Pull request merged**